### ISSUE TITLE
Downloaded or Emailed CSV report columns should better match WooCommerce table in UI

### ISSUE THREAD

**Issue by sverleis** [Automattic/woocommerce-payments#8183]
### Description

When downloading a report from WooPayments, ideally the column structure of the download (or email) should match the values displayed.

### Acceptance criteria

* Report downloads match the WooCommerce table of displayed content
* Ideally, add a checkbox or some form of query to allow download/email of all columns. This may need to be via a modal on the download button, as is done via Stripe Exports.

### Testing instructions

1. Go to _Transactions_.
2. Download the report via the download button.
3. Columns should match what is displayed.
4. To download all column data, allow for some form of selection, as is available with Stripe via the `Export` button: 

<img width="549" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/16252298/ab197d30-6bb0-4440-ba72-f8984575f844">

### Additional context

- Related Ticket(s): 7726672-zen

**Comment by haszari** [comment]
> When downloading a report from WooPayments, ideally the column structure of the download (or email) should match the values displayed.

I agree! @sverleis  can you add a comment (or edit description) to clarify the current buggy behaviour? Helpful to understand what columns are missing/added, in wrong order, etc.

Next steps here would be to test and elaborate what's buggy/broken and then we can better understand priority.

**Comment by glagonikas** [comment]
This came up while trying to export a report for our bookkeepers, that wouldn't include any PII.

So we filtered for the dates we wanted and then hid the columns containing the customer details and hit "download".

I opened the file to check all orders were there and to my surprise, I saw the columns with customer data in it.

**Comment by haszari** [comment]
Thanks for the insights @glagonikas. I can see that it would be helpful if merchants had a way to control the columns included in CSV exports.

**Comment by AashikP** [comment]
Reported in 8642255-zen

While you download a small subset of data, here's the Column headers of the downloaded CSV:

```
Transaction Id | Date / Time | Type | Channel | Paid Currency | Amount Paid | Deposit Currency | Amount | Fees | Net | Order # | Subscription # | Payment Method | Customer | Email | Country | Risk level | Deposit ID | Deposit date | Deposit status
```

With these columns selected:

<img width="359" alt="image" src="https://github.com/user-attachments/assets/f692edf6-92f1-494c-869b-1b701c96f063">


However, when you don't filter the results, and there's more than 25 rows, the CSV is emailed to you, and it contains the following headers (notice the missing Channel header, which is a requirement for merchant in 8642255-zen)

```
Transaction ID | Date / Time (UTC) | Type | Customer Currency | Amount in Customer Currency | Deposit Currency | Amount in Deposit Currency | Fees | Net | Order # | Payment Method | Customer | Email | Country | Risk level | Deposit ID | Deposit date | Deposit status
```

Also note that there isn't an option to show/hide channel, which is not specifically related to this bug.

**One Workaround:** This may not work for bigger stores, but if you're not transacting a lot, the best step here to get the channel field is by filtering using dates to ensure there are no more than 25 results. This will let you download the CSV directly instead of emailing you a link. Example:

<img width="2401" alt="image" src="https://github.com/user-attachments/assets/6e94ed25-ed59-4c79-8bfc-966773815dff">

**Comment by nicdwilson** [comment]
8985102-zd-a8c

**Comment by sverleis** [comment]
Similar issue, where not all fields can be exported for auditing reasons: 9005176-zen

**Comment by Coursados** [comment]
Hi Guys! 

We are a WooPayments customer who is experiencing issues with the core exporting features. WooPayments export for transactions lacking the export functions of Stripe, and we need to contact Woopayments support for the relevant export. 

It would be great to have the same exporting features from Stripe in WooPayments!

Have a great day!

**Comment by Jinksi** [comment]
Note that the WooCommerce → Analytics behaviour is to export all columns in the CSV, regardless of column visibility settings in the UI.

If we change CSV exports to only include visible columns, we should make the same change to Woo core to ensure a consistent user experience.

_To test this, visit Analytics → Orders, hide the 'Customer' column and download: the CSV will include the 'Customer' column._

**Comment by haszari** [comment]
> Note that the WooCommerce → Analytics behaviour is to export all columns in the CSV, regardless of column visibility settings in the UI.
> If we change CSV exports to only include visible columns, we should make the same change to Woo core to ensure a consistent user experience.

Or vice versa, maybe we should always export all columns in WooPayments too. This would be a small feature regression, but:

- Would be simpler system, reduce complexity, less to maintain and explain/document
- CSVs are readily editable, there are ways to remove/hide/reorder/ignore columns in spreadsheet apps or other systems/integrations/scripts
- Would be more consistent with WooCommerce core

With that in mind, I don't think that removing the ability to customise columns would cause major pain/impact for merchants (for the subset that use it, not necessarily all merchants/exports.

**Comment by Jinksi** [comment]
> maybe we should always export all columns in WooPayments too. This would be a small feature regression

This has not yet been implemented for WC or WooPayments – we always export all columns, albeit with inconsistencies across server/client-based exports.

Speaking of regressions, the workaround @AashikP mentioned [above](https://github.com/Automattic/woocommerce-payments/issues/8183#issuecomment-2340250639) to access the `Channel` header may not work once #9969 is resolved via #10049:

> However, when you don't filter the results, and there's more than 25 rows, the CSV is emailed to you, and it contains the following headers (notice the missing Channel header, which is a requirement for merchant in 8642255-zd-a8c)

**Comment by nagpai** [comment]
We have now [moved the CSV exports generation to the Transaction Platform Service](https://github.com/Automattic/woocommerce-payments/issues/9969), through the Simplify CSV Project ( getting launched in v9.0.0 ) . We now have a single way of exports - no more client-based JS export versus Server + email export. 

The missing `Channel` column has also now been included in the exported report column ( via https://github.com/Automattic/transact-platform-server/issues/7140 ) 

With this, the exported columns will match the tables in the UI. We are however currently remaining with exporting all columns, in line with what we have in core 

I am closing this issue for now.

**Comment by glagonikas** [comment]
Hi @nagpai 

> With this, the exported columns will match the tables in the UI. We are however currently remaining with exporting all columns, in line with what we have in core

I'm not sure what you mean by this? Are you saying you'll still force **all** columns on the export? If so, that's not a solution to this issue, as the output still doesn't match the columns viewed on the UI.

I tried with 9.0.0 whilst ensuring the customer data was not dispayed in the UI and the export had all the customer records, so the core issue is not resolved.

It is **imperative** that the UI matches the export, or there's a risk of an accidental leak of PII if people don't check the output file.

**Comment by glagonikas** [comment]
@nagpai can you confirm if this issue is now resolved? We're using 9.3.0 and when exporting the CSV from /wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Ftransactions with the "Customer" and "Email" fields hidden, we still see those columns on the exported CSV.

**Comment by authorshadia** [comment]
Yes, this is resolved. Thank you very much for your help!
Shadia

Bible Teacher | Award-Winning Author | Speaker


New MINI Bible study!
<https://shop.shadiahrichi.com/product/thirst-no-more-bible-study/>
